src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -106,12 +107,14 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
         Phone updatedPhone = editPersonDescriptor.getPhone().orElse(personToEdit.getPhone());
         Email updatedEmail = editPersonDescriptor.getEmail().orElse(personToEdit.getEmail());
         Address updatedAddress = editPersonDescriptor.getAddress().orElse(personToEdit.getAddress());
+        DaysAttended updatedDaysAttended = editPersonDescriptor.getDaysAttended().orElse(personToEdit
+            .getDaysAttended());
         Set<Tag> updatedTags = editPersonDescriptor.getTags().orElse(personToEdit.getTags());
         Set<Subject> updatedSubjects = editPersonDescriptor.getSubjects().orElse(personToEdit.getSubjects());
         Set<String> updatedClasses = editPersonDescriptor.getClasses().orElse(personToEdit.getClasses());
 
-        return new Person(updatedName, updatedGender, updatedPhone, updatedEmail, updatedAddress, updatedTags,
-            updatedSubjects, updatedClasses);
+        return Person.createPerson(personToEdit.getType(), updatedName, updatedGender, updatedPhone, updatedEmail,
```

Comment:
Nice use of a factory method to generalize creation of either Student or Teacher